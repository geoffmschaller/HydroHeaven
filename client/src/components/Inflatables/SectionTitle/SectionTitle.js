import React from 'react';
import PropTypes from 'prop-types';
import styles from './SectionTitle.module.css';

const SectionTitle = (props) => {

	return (
		<div className={styles.sectionTitle}>
			{
				props.darkMode ?
					<>
						<div className={[styles.title, styles.title_dark].join(" ")}>{props.title}</div>
						<div className={[styles.border, styles.border_dark].join(" ")}/>
					</> :
					<>
						<div className={styles.title}>{props.title}</div>
						<div className={styles.border}/>
					</>
			}

		</div>
	);

};

SectionTitle.propTypes = {
	title: PropTypes.string.isRequired,
	darkMode: PropTypes.bool
};

export default SectionTitle;