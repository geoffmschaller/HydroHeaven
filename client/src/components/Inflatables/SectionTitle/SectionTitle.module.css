.sectionTitle {
  width: 100%;
  position: relative;
  margin-bottom: 30px; }
  .sectionTitle .title {
    font-size: 40px;
    background-color: white;
    color: #566573;
    padding: 0 20px;
    position: relative;
    z-index: 5;
    width: 90%;
    max-width: 500px;
    margin: 0 auto;
    text-align: center;
    font-family: 'Poiret One', cursive; }
  .sectionTitle .title_dark {
    background-color: transparent;
    color: #f4f6f6; }
  .sectionTitle .border {
    width: 100%;
    height: 1px;
    background-color: #abb2b9;
    position: absolute;
    top: 50%;
    right: 0;
    left: 0; }
  .sectionTitle .border_dark {
    background-color: transparent; }

@media (max-width: 500px) {
  .sectionTitle .title {
    width: 90%; } }
