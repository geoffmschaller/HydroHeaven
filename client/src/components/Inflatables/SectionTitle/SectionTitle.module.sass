$text-color: #566573
$border-color: #abb2b9

$dark-background: transparent
$dark-text: #f4f6f6
$dark-border: transparent

.sectionTitle
	width: 100%
	position: relative
	margin-bottom: 30px

	.title
		font-size: 40px
		background-color: white
		color: $text-color
		padding: 0 20px
		position: relative
		z-index: 5
		width: 90%
		max-width: 500px
		margin: 0 auto
		text-align: center
		font-family: 'Poiret One', cursive

	.title_dark
		background-color: $dark-background
		color: $dark-text

	.border
		width: 100%
		height: 1px
		background-color: $border-color
		position: absolute
		top: 50%
		right: 0
		left: 0

	.border_dark
		background-color: $dark-border

@media(max-width: 500px)
	.sectionTitle
		.title
			width: 90%