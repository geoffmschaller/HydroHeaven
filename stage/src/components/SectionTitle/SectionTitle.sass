
.section-title
	width: 100%
	padding: 20px 0
	position: relative

	.border
		width: 100%
		height: 1px
		background-color: white