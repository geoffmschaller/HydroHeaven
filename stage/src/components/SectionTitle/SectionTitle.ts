import {Component, Vue} from "vue-property-decorator";

@Component
export default class SectionTitle extends Vue {

}