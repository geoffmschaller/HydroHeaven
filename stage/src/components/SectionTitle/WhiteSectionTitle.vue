<template>
	<div class="section-title">
		<div class="border"/>
		<div class="title">{{title}}</div>
	</div>
</template>

<script src="./SectionTitle.ts"/>
<style scoped lang="sass" src="./SectionTitle.sass"/>