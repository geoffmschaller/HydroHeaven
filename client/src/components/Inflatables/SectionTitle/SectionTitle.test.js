import React from 'react';
import SectionTitle from "./SectionTitle";

const wrapper = <SectionTitle title={"This is the title"}/>;

it("Section Title Inflatable", () => {
	expect(wrapper).toMatchSnapshot();
});