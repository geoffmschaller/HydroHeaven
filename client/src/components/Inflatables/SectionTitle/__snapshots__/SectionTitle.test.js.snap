// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Title Inflatable 1`] = `
<SectionTitle
  title="This is the title"
/>
`;
